cpp/src/arrow/util/CMakeLists.txt

```diff
@@ -48,6 +48,7 @@ add_arrow_test(utility-test
                cache_test.cc
                checked_cast_test.cc
                compression_test.cc
+               counting_semaphore_test.cc
```

Comment:
Perhaps move this into threading-utility-test?

Reply:
Done.